<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SendFrame</class>
 <widget class="QDialog" name="SendFrame">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>668</width>
    <height>862</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SendFrame</string>
  </property>
  <widget class="QPushButton" name="pushButton_send">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>750</y>
     <width>131</width>
     <height>71</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Send</string>
   </property>
  </widget>
  <widget class="QToolBox" name="toolBox">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>20</y>
     <width>631</width>
     <height>651</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="currentIndex">
    <number>1</number>
   </property>
   <widget class="QWidget" name="mac_page">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>631</width>
      <height>329</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
     </font>
    </property>
    <attribute name="label">
     <string>DataLink Layer</string>
    </attribute>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>30</y>
       <width>131</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>source mac:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>170</y>
       <width>111</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>protocol:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>100</y>
       <width>131</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>dest mac:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="ether_type">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>180</y>
       <width>103</width>
       <height>34</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>ARP</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>IP</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Other</string>
      </property>
     </item>
    </widget>
    <widget class="QLineEdit" name="source_mac">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>40</y>
       <width>221</width>
       <height>41</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLineEdit" name="dest_mac">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>110</y>
       <width>221</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="ip_page">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>631</width>
      <height>329</height>
     </rect>
    </property>
    <attribute name="label">
     <string>IP</string>
    </attribute>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>0</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>source IP:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>60</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>dest IP:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="ip_protocol">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>270</y>
       <width>111</width>
       <height>41</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>ICMP</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>TCP</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>UDP</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Other</string>
      </property>
     </item>
    </widget>
    <widget class="QCheckBox" name="ip_df">
     <property name="geometry">
      <rect>
       <x>530</x>
       <y>210</y>
       <width>112</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>DF</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="ip_mf">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>210</y>
       <width>112</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>MF</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="ip_id">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>130</y>
       <width>191</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLineEdit" name="ip_foff">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>200</y>
       <width>191</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLineEdit" name="ip_ttl">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>270</y>
       <width>81</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLineEdit" name="ip_destip">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>60</y>
       <width>191</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLineEdit" name="ip_sourceip">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>0</y>
       <width>191</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_19">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>130</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>ID number</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_20">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>200</y>
       <width>201</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Fragment offset:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_21">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>270</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>TTL:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_22">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>270</y>
       <width>111</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>protocol:</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="arp_page">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>631</width>
      <height>329</height>
     </rect>
    </property>
    <attribute name="label">
     <string>ARP</string>
    </attribute>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>10</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>opcode:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>70</y>
       <width>131</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>source Mac:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>130</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>source IP:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>190</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>dest Mac:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="arp_sourcemac">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>80</y>
       <width>181</width>
       <height>34</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="arp_destmac">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>200</y>
       <width>181</width>
       <height>34</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="arp_sourceip">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>140</y>
       <width>181</width>
       <height>34</height>
      </rect>
     </property>
    </widget>
    <widget class="QComboBox" name="arp_opcode">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>10</y>
       <width>141</width>
       <height>34</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>request(1)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>reply(2)</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>250</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>dest IP:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="arp_destip">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>260</y>
       <width>181</width>
       <height>34</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="icmp_page">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>631</width>
      <height>329</height>
     </rect>
    </property>
    <attribute name="label">
     <string>ICMP</string>
    </attribute>
    <widget class="QLabel" name="label_15">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>20</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>type:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_16">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>80</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>code:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_17">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>150</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Id:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_18">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>220</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>seq:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="icmp_type">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>20</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhDigitsOnly</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="icmp_code">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>80</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhDigitsOnly</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="icmp_id">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>150</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhDigitsOnly</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="icmp_seq">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>220</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhDigitsOnly</set>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="tcp_page">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>631</width>
      <height>329</height>
     </rect>
    </property>
    <attribute name="label">
     <string>TCP</string>
    </attribute>
    <widget class="QLabel" name="label_43">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>source port:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_44">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>0</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>dest port:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_45">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>seq number:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_46">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>60</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>ACK number:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_47">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>130</y>
       <width>161</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>windows size: </string>
     </property>
    </widget>
    <widget class="QLabel" name="label_48">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>130</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>urgent point:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="tcp_sourceport">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>0</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="tcp_destport">
     <property name="geometry">
      <rect>
       <x>450</x>
       <y>0</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="tcp_seqnum">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>60</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="tcp_windowsize">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>130</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="tcp_urgentpoint">
     <property name="geometry">
      <rect>
       <x>450</x>
       <y>130</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="tcp_acknum">
     <property name="geometry">
      <rect>
       <x>450</x>
       <y>60</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QCheckBox" name="tcp_syn">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>200</y>
       <width>112</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>SYN</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="tcp_fin">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>200</y>
       <width>112</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>FIN</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="tcp_ack">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>200</y>
       <width>112</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>ACK</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="tcp_push">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>200</y>
       <width>112</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>PUSH</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="tcp_rst">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>270</y>
       <width>112</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>RST</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="tcp_ecn">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>270</y>
       <width>112</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>ECN</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="tcp_urg">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>270</y>
       <width>112</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>URG</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="tcp_cwr">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>270</y>
       <width>112</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>CWR</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="udp_page">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>631</width>
      <height>329</height>
     </rect>
    </property>
    <attribute name="label">
     <string>UDP</string>
    </attribute>
    <widget class="QLineEdit" name="udp_sourceport">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>33</y>
       <width>91</width>
       <height>41</height>
      </rect>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhDigitsOnly</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="udp_destport">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>120</y>
       <width>91</width>
       <height>41</height>
      </rect>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhDigitsOnly</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>30</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>source port:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>110</y>
       <width>131</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>dest port:</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="data_page">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>631</width>
      <height>329</height>
     </rect>
    </property>
    <attribute name="label">
     <string>Data</string>
    </attribute>
    <widget class="QPlainTextEdit" name="Data">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>0</y>
       <width>551</width>
       <height>301</height>
      </rect>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QLineEdit" name="number">
   <property name="geometry">
    <rect>
     <x>220</x>
     <y>740</y>
     <width>113</width>
     <height>34</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="inputMethodHints">
    <set>Qt::ImhDigitsOnly</set>
   </property>
   <property name="text">
    <string>10</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="time_interval">
   <property name="geometry">
    <rect>
     <x>220</x>
     <y>790</y>
     <width>113</width>
     <height>34</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="inputMethodHints">
    <set>Qt::ImhDigitsOnly</set>
   </property>
   <property name="text">
    <string>500</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_13">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>730</y>
     <width>171</width>
     <height>51</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="text">
    <string>packet number:</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_14">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>780</y>
     <width>191</width>
     <height>51</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="text">
    <string>time interval(ms):</string>
   </property>
  </widget>
  <widget class="QProgressBar" name="progressBar">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>690</y>
     <width>571</width>
     <height>28</height>
    </rect>
   </property>
   <property name="value">
    <number>24</number>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
